programs/sunrise_beam/src/instructions/update_allocations.rs

```diff
@@ -0,0 +1,22 @@
+use anchor_lang::prelude::*;
+
+use crate::state::AllocationUpdate;
+use crate::UpdateBeamAllocations;
+
+pub fn handler(ctx: Context<UpdateBeamAllocations>, values: Vec<AllocationUpdate>) -> Result<()> {
+    let state = &mut ctx.accounts.state;
+
+    for value in values {
+        if let Some(details) = state.get_mut_beam_details(&value.beam) {
+            details.allocation = value.new_allocation;
+        } else {
+            todo!("maybe error?");
```

Comment:
is this triggered if the state does not contain the beam referenced in "values"? Then yes I think it is an error case.

Reply:
I'll update the code to return an `unidentified beam` error here